field/select/classes/field.php

```diff
@@ -502,9 +502,9 @@ public function userform_mform_element($mform, $searchform, $readonly) {
         // End of: element values.
 
         $attributes = array();
-        $attributes['id'] = $idprefix;
```

Comment:
The reason is more than silly. I simply moved the 
$attributes['id'] = $idprefix;
inside the if calculating it only once instead of assigning and then reassigning if.